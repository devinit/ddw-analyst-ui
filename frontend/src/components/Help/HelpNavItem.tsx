import React, { FunctionComponent } from 'react';
import { Dropdown, Nav } from 'react-bootstrap';
import { Route, Switch } from 'react-router-dom';
import { HelpMenu } from '.';
import {
  dataSourcesHelpMenuLinks,
  myDatasetsHelpMenuLinks,
  publishedDatasetsHelpMenuLinks,
  queryBuilderHelpMenuLinks,
  queryDataHelpMenuLinks,
  queryHistoryHelpMenuLinks,
  scheduledEventHelpMenuLinks,
  sourceHistorysHelpMenuLinks,
  updateDataSourceHelpMenuLinks,
} from '../../utils/help';

const HelpNavItem: FunctionComponent = () => {
  return (
    <Dropdown as={Nav.Item} aria-labelledby="navbarDropdownHelp">
      <Dropdown.Toggle as={Nav.Link} id="nav-dropdown" data-cy="help">
        <i className="material-icons">help</i>
        <p className="d-lg-none d-md-block">Help</p>
      </Dropdown.Toggle>
      <Dropdown.Menu alignRight>
        <Switch>
          <Route path="/" exact component={() => <HelpMenu links={myDatasetsHelpMenuLinks} />} />
          <Route
            path="/datasets"
            exact
            component={() => <HelpMenu links={publishedDatasetsHelpMenuLinks} />}
          />
          <Route
            path="/sources"
            exact
            component={() => <HelpMenu links={dataSourcesHelpMenuLinks} />}
          />
          <Route
            path="/queries/build"
            exact
            component={() => <HelpMenu links={queryBuilderHelpMenuLinks} />}
          />
          <Route
            path="/queries/build/:id"
            exact
            component={() => <HelpMenu links={queryBuilderHelpMenuLinks} />}
          />
          <Route
            path="/queries/history/:id"
            exact
            component={() => <HelpMenu links={queryHistoryHelpMenuLinks} />}
          />
          <Route
            path="/update"
            exact
            component={() => <HelpMenu links={updateDataSourceHelpMenuLinks} />}
          />
          <Route
            path="/queries/data/:id"
            exact
            component={() => <HelpMenu links={queryDataHelpMenuLinks} />}
          />
          <Route
            path="/scheduledevents"
            exact
            component={() => <HelpMenu links={scheduledEventHelpMenuLinks} />}
          />
          <Route
            path="/source/datasets/:id"
            exact
            component={() => <HelpMenu links={publishedDatasetsHelpMenuLinks} />}
          />
          <Route
            path="/source/history/:id"
            exact
            component={() => <HelpMenu links={sourceHistorysHelpMenuLinks} />}
          />
        </Switch>
      </Dropdown.Menu>
    </Dropdown>
  );
};

export { HelpNavItem };
