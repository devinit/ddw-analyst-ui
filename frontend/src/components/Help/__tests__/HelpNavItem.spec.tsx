import 'jest-styled-components';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { HelpNavItem } from '../HelpNavItem';

describe('HelpNavItem', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<HelpNavItem />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
