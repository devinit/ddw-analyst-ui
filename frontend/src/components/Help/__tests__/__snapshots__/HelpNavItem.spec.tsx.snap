// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpNavItem renders correctly 1`] = `
<div
  aria-labelledby="navbarDropdownHelp"
  className="dropdown nav-item"
  onKeyDown={[Function]}
>
  <a
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle nav-link"
    data-cy="help"
    href="#"
    id="nav-dropdown"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <i
      className="material-icons"
    >
      help
    </i>
    <p
      className="d-lg-none d-md-block"
    >
      Help
    </p>
  </a>
</div>
`;
